code/game/gamemodes/events/powerfruit.dm

```diff
@@ -0,0 +1,90 @@
+//the actual powerfruit obj
+#define POWER_PER_FRUIT 500
+#define SPREAD_CHANCE 15
+#define ATTACK_CHANCE 15
+
+/obj/structure/cable/powerfruit
+	name = "powerfruit"
+	desc = "A strange alien fruit somehow creating power. How curious! It looks dangerous, better not touch it"
+	icon = 'icons/obj/lighting.dmi' //TODO
+	icon_state = "glowshroomf" //TODO
+	level = LEVEL_ABOVE_FLOOR
+	plane = ABOVE_HUMAN_PLANE
+	pass_flags = PASSTABLE | PASSGRILLE | PASSGIRDER | PASSMACHINE
+
+/obj/structure/cable/powerfruit/New(loc)
+	var/datum/powernet/PN = getFromPool(/datum/powernet)
+	PN.add_cable(src)
+
+	for(var/dir in cardinal)
+		mergeConnectedNetworks(dir)   //Merge the powernet with adjacents powernets
+	mergeConnectedNetworksOnTurf() //Merge the powernet with on turf powernets
+
+	processing_objects += src
+	. = ..()
+
+/obj/structure/cable/powerfruit/Destroy()
+	processing_objects -= src
+	..()
+
+/obj/structure/cable/powerfruit/process()
+	//add power to powernet
+	add_avail(POWER_PER_FRUIT)
+
+	//spread - copypasta from spreading_growth.dm
+	var/list/neighbors = getViableNeighbours()
```

Comment:
You don't need this list unless the probability below succeeds. Move it to within the if statement